src/lib/proof_system.ts

```diff
@@ -578,6 +587,33 @@ async function compileProgram({
   let maxProofs = getMaxProofsVerified(methodIntfs);
   overrideWrapDomain ??= maxProofsToWrapDomain[maxProofs];
 
+  let picklesCache: Pickles.Cache = [
+    0,
+    function read_(mlHeader) {
+      let header = parseHeader(proofSystemTag.name, methodIntfs, mlHeader);
+      try {
+        let bytes = cache.read(header);
+        if (bytes === undefined) return MlResult.unitError();
+        return MlResult.ok(decodeProverKey(mlHeader, bytes));
+      } catch (e: any) {
+        return MlResult.unitError();
+      }
+    },
```

Comment:
this code connects all the dots:

* our helpers to create cache headers and decode snark keys from bytes
* the user-defined `cache` where we can `read` and `write` from
* the `Pickles.Cache` which Pickles expects